src/node-saml/saml.ts

```diff
@@ -1201,7 +1219,11 @@ class SAML {
     if (notOnOrAfter) {
       const notOnOrAfterMs = Date.parse(notOnOrAfter);
       if (nowMs - this.options.acceptedClockSkewMs >= notOnOrAfterMs)
-        return new Error("SAML assertion expired");
+        return new Error("SAML assertion expired: clocks skewed too much");
+    }
+    if (maxTimeLimitMs) {
+      if (nowMs - this.options.acceptedClockSkewMs >= maxTimeLimitMs)
```

Comment:
I note that, in connection with my previous comment, there is likely a bug here. If someone corrupts the `notOnOrAfter` or `notBeforeMs`, or there is some parsing issue with them, then we'll never fail a date check because we never check for `NaN` and a comparison with `NaN` is always false, so we'll never throw.  Is that how you read it?

Reply:
That is indeed how I read it. 

To clarify, this is an existing bug, right? Not one this PR introduces? 

(I'm happy to fix it here though, nice catch!)

Reply:
It is an existing bug, but your changes would propagate it. Thanks for fixing it!